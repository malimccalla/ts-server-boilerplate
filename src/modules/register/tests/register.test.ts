import { gql } from 'apollo-server';
import { request } from 'graphql-request';
import { host } from '../../../constants';

const registerMutation = (email: any, password: any) => gql`
    mutation {
      register(email: "${email}", password: "${password}") {
        ok
        user {
          id
          email
        }
        errors {
          path
          message
        }
      }
    }
  `;

describe('Register', () => {
  const validEmail = 'tester@mali.com';
  const validPassword = 'superS3Cr7t!';

  const invalidEmail = 'Invalid email';
  const invalidPassword = 'pass123';

  describe('Email validation', () => {
    test('should not allow an invalid email', async () => {
      const mutation = registerMutation(invalidEmail, validPassword);
      const { register } = (await request(host, mutation)) as {
        register: GQL.IRegisterResponse;
      };

      expect(register.ok).toBe(false);
      expect(register.errors).toHaveLength(1);
      expect(register.user).toBeFalsy();
      expect(register.errors).toMatchSnapshot();
    });
  });

  describe('Password validation', () => {
    test('should not allow a password less than 8 characters', async () => {
      const mutation = registerMutation(validEmail, invalidPassword);
      const { register } = (await request(host, mutation)) as {
        register: GQL.IRegisterResponse;
      };

      expect(register.ok).toBe(false);
      expect(register.errors).toHaveLength(1);
      expect(register.user).toBeFalsy();
      expect(register.errors).toMatchSnapshot();
    });
  });
});
