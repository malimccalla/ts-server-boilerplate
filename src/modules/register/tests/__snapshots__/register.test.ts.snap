// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Email validation should not allow an invalid email 1`] = `
Array [
  Object {
    "message": "Invalid email",
    "path": "email",
  },
]
`;

exports[`Register Password validation should not allow a password less than 8 characters 1`] = `
Array [
  Object {
    "message": "Password must be longer than 8 characters",
    "path": "password",
  },
]
`;
